### PR TITLE
create class Address and override method Equals(Address).

### PR DESCRIPTION
**Pull request by NadezhdaPrilepa**
Create class Address and override method Equals(Address).Pass test with Junit (100% coverage)

### CODE REVIEW

File: teach/src/main/java/skype/teach/np/NpAddress.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (!(obj instanceof Address)) {
+            return false;
+        }
+        Address addr = (Address) obj;
+        // indexNum not important for result
+        if (this.cityName.equals(addr.cityName)) {
```
Comment (vbogdanovsky) [inline teach/src/main/java/skype/teach/np/NpAddress.java]: Any field can be unset. Valid address may contain only city.
So equals should return true if street is null in both addresses, for example.

File: teach/src/test/java/skype/teach/np/NpAddressTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        Address adr9 = new Address("Moscow", "stoleshnikov", "p", 23, 34, 1234);
+        Object obj = new Object();
+
+        // test different flatNum
+        assertEquals("adr.equals(adr1)", false, adr.equals(adr1));
+        // test the same Address
+        assertEquals("adr.equals(adr)", true, adr.equals(adr));
+        // test adr=null
+        assertEquals("adr.equals(null)", false, adr.equals(null));
+        //test different instance of object
```
Comment (vbogdanovsky) [inline teach/src/test/java/skype/teach/np/NpAddressTest.java]: Such comments should be added to assertEquals messages.

File: teach/src/main/java/skype/teach/np/NpAddress.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (!(obj instanceof Address)) {
+            return false;
```
Comment (vbogdanovsky) [inline teach/src/main/java/skype/teach/np/NpAddress.java:33]: I don't see a test case for this line.

Comment (NadezhdaPrilepa) [inline teach/src/main/java/skype/teach/np/NpAddress.java:33]: In NpAddressTest 40 line compared Address object and Object

**Pull request merged**